### DISCUSSION TITLE
エフェクトシステム

### DISCUSSION THREAD

**Discussion by Suyarumi** [TUSB/TheUnusualSkyBlock#95] [各種システムまとめ]
エフェクト全般の管理用のシステムのまとめ

**Comment by Suyarumi** [comment]
特殊エフェクト一覧
死の宣告(10カウント)
死の宣告(3カウント)
恐怖
病気
混乱
トント
衰弱
火傷
エーテルなどのMP回復エフェクト
解毒薬(バフデバフ全消し) ノンデメリット(デバフ全消し)

**Comment by Suyarumi** [comment]
特殊エフェクトの効果
死の宣告(10カウント)
10カウント(30秒)後に死亡。
聖水を飲むことで治癒。
死の宣告(3カウント)
3カウント(10秒)後に死亡。
聖水を飲むことで治癒。
恐怖
プレイヤーのスキルを確率で妨害する。
時間経過で治癒。
付与する時に治癒までの時間とスキルの妨害確率を指定できるようにする。
病気
プレイヤーの体力回復を確率で妨害する。
時間経過で治癒。
付与する時に治癒までの時間と妨害確率を設定できるようにする。
混乱
エフェクト効果時間中一定確率で真後ろを向く。
時間経過で治癒
付与する時に治癒までの時間と真後ろを向く確率を設定できるようにする。
トント
一定回数ダメージを受けると爆発する。
時間経過で治癒はしない。
付与する時に爆発までのカウントと爆発時のダメージか爆発半径を設定できるようにする。
火傷
炎上ダメージを増加させる。炎上状態の時以外は付与されない。
プレイヤーの炎上状態が解除されたら治癒。
炎上時の追加ダメージを増減できるようにする。
衰弱
HPを1列(20)まで強制的に減らす。
エーテル(MP回復固定値)
プレイヤーのMPを固定値で回復する。
MPの回復量を設定できるようにする。
エーテル(MP回復割合)
プレイヤーのMPを割合で回復する。
MPの回復量を設定できるようにする。

**Comment by anminmakura** [comment]
##############################
### 特殊効果適用
##############################

### invisibility 1 1
死の宣告
### invisibility 1 2
死の宣告3カウントバージョン
### invisibility 1 3
やけど
### invisibility 1 4
衰弱(ハート1列)
### invisibility 1 5
病気
### invisibility 1 6
混乱
### invisibility 1 7
呪詛
### invisibility 1 8
恐怖
### invisibility 1 9
トント

### invisibility 1 11
MP25回復
### invisibility 1 12
MP50回復
### invisibility 1 13
MP100回復
### invisibility 1 14
MP200回復

### invisibility 1 21
ハート2.5回復
### invisibility 1 22
ハート5回復
### invisibility 1 23
ハート10回復
### invisibility 1 24
ハート20回復

### invisibility 1 31
ハート2.5回復・MP25回復
### invisibility 1 32
ハート5回復・MP50回復
### invisibility 1 33
ハート10回復・MP100回復
### invisibility 1 34
ハート20回復・MP200回復

### invisibility 1 41
MP1割回復
### invisibility 1 42
MP3割回復
### invisibility 1 43
MP6割回復
### invisibility 1 44
MP全回復

### invisibility 1 51
HP2割回復
### invisibility 1 52
HP6割回復
### invisibility 1 53
HP12割回復
### invisibility 1 54
HP20割回復

### invisibility 1 61
MP1割回復・HP2割回復
### invisibility 1 62
MP3割回復・HP6割回復
### invisibility 1 63
MP6割回復・HP12割り回復
### invisibility 1 64
MP全回復・HP20割回復

### invisibility 1 71
聖水
### invisibility 1 73
やけど回復
    ### invisibility 1 74
    衰弱回復(保留)
### invisibility 1 75
病気回復
### invisibility 1 76
混乱回復
### invisibility 1 77
呪詛回復
### invisibility 1 78
恐怖回復
### invisibility 1 79
トント回復

### invisibility 1 81
免疫 20%
### invisibility 1 82
免疫 40%
### invisibility 1 83
免疫 60%
### invisibility 1 84
免疫 80%

### invisibility 1 121
パナケイア(全デバフ消去)
### invisibility 1 122
ソーマ(HPMP全回復+耐性100)
### invisibility 1 123
炎上
### invisibility 1 124
スキル再設定可能

## 保留
_### health_boost <ダメージ量> 0
MOBのみにダメージを発生させる
instant_damage 127レベル Duration 1と instant_health 127レベル Duration 1と health_boost 0 レベル Duration=ダメージ量 を同時に付与する_

**Comment by anminmakura** [comment]
### 特殊エフェクトシリーズのバフ効果にavoid void(奈落復帰)を新たに製作予定
効果は奈落に入ると1度だけ強烈な浮遊がかかるというもの。
予定として通常のTUSB神社に正常にお参りすることで付与される。
(賽銭箱に特定のアイテム投げることで付与予定)

function名とか内部名はavoid_void
